src/project.rs

```diff
@@ -320,6 +328,39 @@ impl FuzzProject {
         }
     }
 
+    // Helper function for `exec_build()` used to expose cargo-fuzz information
+    // via environment variables. Such environment variables can be used by fuzz
+    // target dependencies' build scripts to detect whether or not cargo-fuzz is
+    // responsible for the build.
+    //
+    // This is called directly before the `cargo build ...` command is executed.
+    fn build_env_expose(
+        &self,
+        _mode: options::BuildMode,
+        _build: &options::BuildOptions,
+        _fuzz_target: Option<&str>,
+    ) -> Result<()> {
+        // expose a flag environment variable to allow the detection of cargo-fuzz
+        env::set_var(BUILD_ENV_FLAG, "1");
+
+        // expose the path to the cargo-fuzz manifest
+        let manifest_path = self.manifest_path();
+        env::set_var(BUILD_ENV_MANIFEST_DIR, manifest_path.as_os_str());
```

Comment:
Rather than setting the env var in this process's environment, which then require unsetting them later, can this method take a `&mut std::process::Command` argument and attach the new env vars only to the subprocess's environment via the [`Command::env`](https://doc.rust-lang.org/nightly/std/process/struct.Command.html#method.env) method?